python/cudf/cudf/core/frame.py

```diff
@@ -458,6 +458,61 @@ def reassign_categories(categories, cols, col_idxs):
 
         return out
 
+    def equals(self, other, **kwargs):
+        """
+        Test whether two objects contain the same elements.
+        This function allows two Series or DataFrames to be compared against
+        each other to see if they have the same shape and elements. NaNs in
+        the same location are considered equal. The column headers do not
+        need to have the same type.
+
+        Parameters
+        ----------
+        other : Series or DataFrame
+            The other Series or DataFrame to be compared with the first.
+
+        Returns
+        -------
+        bool
+            True if all elements are the same in both objects, False
+            otherwise.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> s = cudf.Series([1, 2, 3])
+        >>> other = cudf.Series([1, 2, 3])
+        >>> s.equals(other)
+        True
+        >>> different = cudf.Series([1.5, 2, 3])
+        >>> s.equals(different)
+        False
+        """
+        if self is other:
+            return True
+
+        if other is None or len(self) != len(other):
+            return False
+
+        if kwargs.get("check_types", True):
+            if type(self) is not type(other):
+                return False
+
+        # check data:
+        for self_col, other_col in zip(
+            self._data.values(), other._data.values()
+        ):
+            if not self_col.equals(other_col):
+                return False
+
+        # check index:
+        if self._index is None:
+            return other._index is None
+        else:
+            if other._index is None:
+                return False
```

Comment:
Shouldn't this just be handled in `Index.equals`?

Reply:
Do you mean the `else` bit?

Reply:
Yes, apologies, highlighted more than necessary.

Reply:
Fixed